commcare_connect/microplanning/management/commands/generate_microplanning_data.py

```
@@ -0,0 +1,584 @@
+        """The caller's existing opportunity, or the demo one (created on first run).
+
+        Returns the opportunity and whether its workspace is one this command owns — a caller's
+        own workspace is only ever read, never reconfigured.
+        """
+        if options["opp_id"]:
+            try:
+                opportunity = get_object_by_uuid_or_int(
+                    Opportunity.objects.all(), options["opp_id"], uuid_field="opportunity_id"
+                )
+            except Http404:
+                raise CommandError(f"No opportunity matches --opp-id {options['opp_id']!r}")
```

Ruff (0.16.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)